stackrox/stackrox
# ROX-19706: Add init-bundles to clusters route

## Description

1. Follow procedure in https://github.com/stackrox/stackrox/tree/master/ui/apps/platform#add-a-feature-flag-to-frontend-code
2. Make request in InitBundlesRoute.tsx file and render one of the following:
    * InitBundleForm.tsx
    * InitBundleView.tsx
    * InitBundlesTable.tsx

### Residue

1. Write procedure to add a route.
2. Add callbacks and form.
3. Write integration tests.
4. Collaborate with technical docs team to improve explanation after creating a bundle.

## Checklist

## Testing Performed
1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 427 = 3764386 - 3763959
        total: 8319 = 10076764  - 10068445
    * `ls -al build/static/js/*.js | wc -l`
        0 = 81 - 81 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/integrations/authProviders/clusterInitBundles
    * See classic table with name column only.
        ![integrations_authProviders_clusterInitBundle](https://github.com/stackrox/stackrox/assets/11862657/73f1cb17-9f8a-4d69-8942-1210999b3c14)

2. Click the link
    * See classic details view.
        ![integrations_authProviders_clusterInitBundle_id](https://github.com/stackrox/stackrox/assets/11862657/d377ff87-158b-4ff5-8f89-527c04b7531d)

3. Visit /main/clusters/init-bundles **without** ROX_MOVE_INIT_BUNDLES_UI feature flag
    * See 500 status for bogus GET /v1/clusters/init-bundles request
        ![clusters_init-bundles_500](https://github.com/stackrox/stackrox/assets/11862657/11802bf5-a46d-46a0-82d0-4125031b5b89)

4. Visit /main/clusters/init-bundles **with** ROX_MOVE_INIT_BUNDLES_UI feature flag
    * See spinner with temporary code change not to reset `isLoading` state.
        ![clusters_init-bundles_200_isLoading](https://github.com/stackrox/stackrox/assets/11862657/cc005288-61e7-4fc4-886d-48f48bdd5f1f)

    * See error message with temporary code change to mangle API endpoint
        ![clusters_init-bundles_404](https://github.com/stackrox/stackrox/assets/11862657/ac46b32e-74cc-42a6-b790-0e20da167e12)

    * See init bundles table
        ![clusters_init-bundles_200](https://github.com/stackrox/stackrox/assets/11862657/473e0c7b-9d3c-4408-afff-cc9c879fa5da)

    * Click link to see init bundle view
        ![clusters_init-bundles_id](https://github.com/stackrox/stackrox/assets/11862657/3b0efe32-09b7-4b48-bbd3-9c16bec0fd10)

    * Edit page address to see **Unable to find init bundle**
        ![clusters_init-bundles_id_Unable_to_find](https://github.com/stackrox/stackrox/assets/11862657/bb4df304-1317-425b-a6b8-4a4e3ec848f3)
